adds vanilla metricbeat (#526)
logzio/logz-docs: _source/logzio_collections/_metrics-sources/ec2.md

**Recommended version**: Metricbeat version 7.5.x

#### Configuration

{% include trust-chain-warning.html msg='docker' %}

<div class="tasklist">

##### Set up your IAM user

You'll need an [IAM user](https://console.aws.amazon.com/iam/home)
with these permissions:

* `cloudwatch:GetMetricData`
* `cloudwatch:ListMetrics`
* `ec2:DescribeInstances`
* `ec2:DescribeRegions`
* `iam:ListAccountAliases`
* `sts:GetCallerIdentity`

If you don't have one, set that up now.

Create an **Access key ID** and **Secret access key** for the IAM user,
and paste them in your text editor.

You'll need these for your Metricbeat configuration later.

##### Get your metrics region

You'll need to specify the AWS region you're collecting metrics from.

![AWS region menu](https://dytvr9ot2sszz.cloudfront.net/logz-docs/aws/region-menu.png)

Find your region's slug in the region menu
(in the top menu, on the right side).

For example:
The slug for US East (N. Virginia)
is "us-east-1",
and the slug for Canada (Central) is "ca-central-1".

Paste your region slug in your text editor.
You'll need this for your Metricbeat configuration later.

> ##### Download the Logz.io public certificate to your Metricbeat server

Reviewer: Not 100% sure if this section needs to change because of recent certificate events

{% include trust-chain-warning.html %}

For HTTPS shipping, download the Logz.io public certificate to your certificate authority folder.

```shell
sudo curl https://raw.githubusercontent.com/logzio/public-certificates/master/TrustExternalCARoot_and_USERTrustRSAAAACA.crt --create-dirs -o /etc/pki/tls/certs/COMODORSADomainValidationSecureServerCA.crt
```

##### _(Optional)_ Disable the system module

By default, Metricbeat ships system metrics from its host.
If you don't need these metrics,
disable the system module:

```shell
sudo metricbeat modules disable system
```

##### Configure Metricbeat

If you're working with the default configuration file,
(`/etc/metricbeat/metricbeat.yml`), clear the contents and start with a fresh file.

This code block lays out the default options
for collecting metrics from EC2.
Paste the code block.
Then adjust it to match your AWS environment.

```yml
metricbeat.modules:
- module: aws
  period: 300s
  metricsets:
    - cloudwatch
  metrics: #specify aws namespaces you want to monitor, just add namspaces from AWS list

>     - namespace: <<namespace>>

Reviewer: - namespace: AWS/EC2

>     - namespace: <<namespace>>

Author: ```suggestion
    - namespace: AWS/EC2
```

  access_key_id: '<<access_key_id>>'
  secret_access_key: '<<secret_access_key>>'

> 

Author: ```suggestion
default_region: <<YOUR-AWS-REGION>>

```

> 

Reviewer: default_region: <<YOUR-AWS-REGION> we can add this

fields:
  logzio_codec: json
  token: <<SHIPPING-TOKEN>>
fields_under_root: true
ignore_older: 3hr
type: metrics
output.logstash:
  hosts: ["<<LISTENER-HOST>>:5015"]

>   ssl.certificate_authorities: ['/etc/pki/tls/certs/COMODORSADomainValidationSecureServerCA.crt']

Reviewer: I'm not 100% sure if the certificate name "COMODORSADomainValidationSecureServerCA" is going to change because of recent events or not. do you know?

```

##### Replace the placeholders in your Metricbeat configuration

Still in the same configuration file, replace the placeholders to match your specifics.

* {% include log-shipping/replace-vars.html token=true %}

* {% include log-shipping/replace-vars.html listener=true %}

> * Replace the namespaces you want to ship from. You can refer to this [resource](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/aws-services-cloudwatch-metrics.html) for the complete list of valid AWS namespaces.

Reviewer: We can remove this line. because this page is specific to ec2 only

Reviewer: Replace the namespaces you want to ship from. You can refer to this [resource](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/aws-services-cloudwatch-metrics.html) for the complete list of valid AWS namespaces.

> * Replace the namespaces you want to ship from. You can refer to this [resource](https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/aws-services-cloudwatch-metrics.html) for the complete list of valid AWS namespaces.

Author: ```suggestion

```

One last validation - make sure Logz.io is the only output and appears only once.
If the file has other outputs, remove them.

##### Monitor advanced EC2 metrics (_Optional_)

If you want, you can also monitor advanced EC2 metrics, including disk memory and swap memory. 

> To enable this option, you’ll need to install and configure a CloudWatch agent on your machine and specify the **CWAgent** namespace under the AWS_NAMESPACES parameter. For example:

Reviewer: AWS_NAMESPACES parameter => metrics section

> To enable this option, you’ll need to install and configure a CloudWatch agent on your machine and specify the **CWAgent** namespace under the AWS_NAMESPACES parameter. For example:

Author: ```suggestion
To enable this option, you’ll need to install and configure a CloudWatch agent on your machine and specify the **CWAgent** namespace under the metrics section. For example:
```

{% include trust-chain-warning.html %}

For HTTPS shipping, download the Logz.io public certificate to your certificate authority folder.

```shell
sudo curl https://raw.githubusercontent.com/logzio/public-certificates/master/TrustExternalCARoot_and_USERTrustRSAAAACA.crt --create-dirs -o /etc/pki/tls/certs/COMODORSADomainValidationSecureServerCA.crt
```

##### _(Optional)_ Disable the system module

By default, Metricbeat ships system metrics from its host.
If you don't need these metrics,
disable the system module:

```shell
sudo metricbeat modules disable system
```

##### Configure Metricbeat

If you're working with the default configuration file,
(`/etc/metricbeat/metricbeat.yml`), clear the contents and start with a fresh file.

This code block lays out the default options
for collecting metrics from EC2.
Paste the code block.
Then adjust it to match your AWS environment.

```yml
metricbeat.modules:
- module: aws
  period: 300s
  metricsets:
    - cloudwatch
  metrics: #specify aws namespaces you want to monitor, just add namspaces from AWS list
    - namespace: AWS/EC2

  access_key_id: '<<access_key_id>>'
  secret_access_key: '<<secret_access_key>>'

> default_region: <<YOUR-AWS-REGION>>

Author: ```suggestion
  default_region: <<YOUR-AWS-REGION>>
```

> default_region: <<YOUR-AWS-REGION>>

Author: ```suggestion
```
